src/js/medium-editor.js

```diff
@@ -1220,6 +1220,11 @@ else if (typeof define === 'function' && define.amd) {
 
                     self.createLink(this, target, button);
                 }
+                else if (e.keyCode === 27) {
+                    e.preventDefault();
+                    self.showToolbarActions();
+                    restoreSelection.call(self, self.savedSelection);
```

Comment:
I think ESC should cancel and not create the link thats the usualy standard for dialog behavior for that key... 

From:
http://access.aol.com/dhtml-style-guide-working-group/#dialogmodal
"ESC There should be a method to close the dialog without taking any action."